dd-java-agent/instrumentation/servlet/request-3/src/main/java/datadog/trace/instrumentation/servlet3/Servlet3Advice.java

```diff
@@ -82,6 +82,7 @@ public static void stopSpan(
         if (resp.getStatus() == HttpServletResponse.SC_OK) {
           // exception is thrown in filter chain, but status code is incorrect
           span.setTag(Tags.HTTP_STATUS, 500);
+          span.setError(true);
```

Comment:
If you move this logic to `onResponse` in `HttpClientDecorator`, it should remove the duplication across all the instrumentation.

Reply:
now that I think about it, maybe that won't work because it requires both the response and the exception... never mind.  Maybe a different method that takes both?

Reply:
Are you good with pushing this change to #12? We'll need to revisit status mapping there, I'll keep this in mind.